### PR TITLE
Sysadmin add progess tasks to kill table

### PR DESCRIPTION
**Pull request by caseylitton**
@dcadams @stvstnfrd @caesar2164 Please review PR for including 'Progress' task state alongside 'Queuing' in the sysadmin 'Instructor Task' tab.  Previously only Queue state tasks could be found and killed in the 'Queuing Task' tab.  Now Progress tasks are included in the query and are available in the table.  The kill submission function needed to be updated to allow killing progress task also.

### CODE REVIEW

**Comment by dcadams** [comment]
With a job being in "PROGRESS" it's more likely that the task itself will still be running. What happens, in such a case, if you update the db state to "FAILURE" and try to run the same job again?

File: lms/djangoapps/dashboard/sysadmin.py [hunk 1]
```diff
@@ -929,7 +929,10 @@ def kill_task(self, row_id=None):
         try:
             task = InstructorTask.objects.get(
                 id=row_id,
-                task_state='QUEUING',
+                task_state__in = [
```
Comment (caesar2164) [inline lms/djangoapps/dashboard/sysadmin.py]: why the change from `task_state` to `task_state__in`?

Comment (caesar2164) [inline lms/djangoapps/dashboard/sysadmin.py]: also `task_state__in` has 2 underscores...

Comment (stvstnfrd) [inline lms/djangoapps/dashboard/sysadmin.py]: @caesar2164 https://docs.djangoproject.com/ja/1.9/topics/db/queries/#retrieving-specific-objects-with-filters

Comment (caesar2164) [inline lms/djangoapps/dashboard/sysadmin.py]: cool

**Comment by caesar2164** [comment]
I got nothing else: 👍

File: lms/templates/sysadmin_dashboard.html [hunk 2]
```diff
@@ -63,7 +63,7 @@ <h2 class="instructor-nav">
       <a href="${reverse('sysadmin_staffing')}" class="${modeflag.get('staffing')}">${_('Staffing and Enrollment')}</a>
       ## Translators: refers to http://git-scm.com/docs/git-log
       <a href="${reverse('gitlogs')}">${_('Git Logs')}</a>
-      <a href="${reverse('sysadmin_task_queue')}" class="${modeflag.get('task_queue')}">${_('Task Queue')}</a>
+      <a href="${reverse('sysadmin_task_queue')}" class="${modeflag.get('task_queue')}">${_('Instructor Tasks')}</a>
```
Comment (stvstnfrd) [inline lms/templates/sysadmin_dashboard.html]: Nit: I'm a fan of "less is more"; 
why not just "Tasks"?

File: lms/templates/sysadmin_dashboard.html [hunk 3]
```diff
@@ -189,7 +189,7 @@ <h2 class="instructor-nav">
 %endif
 
 %if modeflag.get('task_queue'):
-  <h3>${_('Kill Queuing Tasks')}</h3><br/>
+  <h3>${_('Kill Progess or Queuing Task')}</h3><br/>
```
Comment (stvstnfrd) [inline lms/templates/sysadmin_dashboard.html]: Per above: would "Kill Tasks" be more straight forward?

File: lms/djangoapps/dashboard/sysadmin.py [hunk 4]
```diff
@@ -969,7 +975,7 @@ def make_datatable(self):
                 _('Task Type'),
                 _('Task State'),
             ],
-            title=_('List of Queueing Tasks'),
+            title=_('List of Tasks'),
```
Comment (stvstnfrd) [inline lms/djangoapps/dashboard/sysadmin.py]: Yeah, I like the brevity here.

**Comment by stvstnfrd** [comment]
@caseylitton Thanks for the pylint fix. I have a few nitpicks about language,
but otherwise this looks good.

File: lms/templates/sysadmin_dashboard.html (hunk 3 above)
Comment (caseylitton) [inline lms/templates/sysadmin_dashboard.html]: "Set task state to failure" would be more clear.

Comment (stvstnfrd) [inline lms/templates/sysadmin_dashboard.html]: > "Set task state to failure" would be more clear.

I disagree; that's leaking implementation details.

Beyond the verbosity, if edX later changes the underlying implementation [1],
maybe they change the name of the failed/canceled state or some other
details, then this becomes a maintenance burden. Now we have to keep
user-facing documentation in-sync with details of the technical
implementation. Then someone goes to "set task state to failure" but
can't find that option because now the text has been changed again.
Spoiler Alert: edX already has plans to rework this.

That the way to "kill" a task here happens to be a specific database
modification shouldn't be of immediate concern to the user.
I agree that "kill" could possibly be misleading without context; it's
admittedly not obvious that it doesn't halt a running process.
But as it were, "kill task" is the colloquialism that our team has
historically used.

Here are some snippets from the last email that Monica sent regarding
this issue:

> Also, since grade report activities are not displaying in the list of
> Task Queues, there isn't any way for us to kill this task.

and

> Once this failed task is killed [...]

Seeing as this is how we've always described this action, my preference
is to not introduce new vernacular at this point.

[1] https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/edx-code/17DA6dt2Dmc/cdoFMMinBAAJ

**Comment by caseylitton** [comment]
@stvstnfrd @caesar2164 Please double check label changes in lms/templates/sysadmin_dashboard.html and I'll squash.

**Comment by stvstnfrd** [comment]
+1

**Comment by caesar2164** [comment]
👍
